aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -117,9 +120,39 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (exceptionThrown) {
         io().deleteFile(newMetadataLocation);
       }
+
+      unlock();
+    }
+  }
+
+  private void acquireLock() {
```

Comment:
Why does this throw `CommitFailedException` when the lock timeout expires and on any exception thrown by the lock manager?

That will cause a retry, which will try to acquire the lock again. I don't think that's the right behavior because failure to get the lock has already exceeded the lock retry timeout. To wait longer, just extend the lock timeout.

Reply:
This is also following the behavior of `HiveTableOperations`. I think it makes sense to throw `CommitFailedException` because locking is a part of `doCommit`, so the failure to get lock is a type of commit failure. In the approach you described, the wait time can be long, and in the current approach the wait time is short for each retry.